### ISSUE TITLE
Pathological inputs take extremely long to compute

### ISSUE THREAD

**Issue by thoppe** [mattdesl/atcq#2]
I'm using the library to generate a palette for every image in a movie. For most images with a diverse palette the computation time is 4-5 seconds. However, for images with very little variation, like the credits at the end of a movie, the computation can take a minute or more. Is there a way to shortcut the algorithm if it can't build a proper palette?

```
>> time nodejs extract.js baboon.png 

real	0m4.877s
user	0m5.145s
sys	0m0.102s

>> time nodejs extract.js 00000000.jpg 
[{"weight":1,"hex":"#000000","R":0,"G":0,"B":0}]

real	1m41.776s
user	1m36.971s
sys	0m0.092s
```

The input image is attached here, the worst-case scenario for an all black screen

![00000000](https://user-images.githubusercontent.com/2707106/80842784-618b0c00-8bb7-11ea-913f-b39689d7cf1d.jpg)

Experimenting with `maxColors` or `targetColors` doesn't seem to make a difference in the output.

**Comment by thoppe** [comment]
For what it's worth, about 5000 frames for each second of the movie _Zoolander_ look like this when processed by the algorithm:

![zoolander_example](https://user-images.githubusercontent.com/2707106/80851522-2ac4ee00-8bd7-11ea-9cb9-41fce24774b6.png)

The bulk of the computation (almost 90% of it!) is spent on the darkest 5% of the frames.

**Comment by mattdesl** [comment]
Should be fixed in `atcq@1.0.5`, I changed `dist < T` to `dist <= T` in one of the threshold conditions. This differs from the paper (presumably the paper has never tested against a pure black or near-black image) but in my very quick tests it doesn't seem to have any negative impact on the algorithm.

In `test/test-black.js` you can see some settings I've applied that produce accurate outputs pretty quickly.

**Comment by thoppe** [comment]
This is a huge improvement for those nearly monochromatic images, take a look at the timing now!

```
nodejs extract.js 00000000.jpg  5 32
[{"weight":1,"hex":"#000000","R":0,"G":0,"B":0}]

real	0m0.661s
user	0m0.716s
sys	0m0.021s
```

Thanks for the quick fix.
